mod-source-record-manager-server/src/main/java/org/folio/verticle/consumers/util/MarcImportEventsHandler.java

```diff
@@ -172,8 +172,19 @@ private Future<JournalRecord> populateRecordTitleIfNeeded(JournalRecord journalR
             })
             .orElseGet(() -> Future.succeededFuture(journalRecord)));
       }
+    } else if (entityType == PO_LINE) {
+      String recordAsString = eventPayload.getContext().get(entityType.value());
+      if (StringUtils.isNotBlank(recordAsString)) {
+        var title = Json.decodeValue(recordAsString, RelatedPoLineInfo.class)
```

Comment:
propose to create JsonObject from PO_LINE value and use it to extract PO_LINE_TITLE field because RelatedPoLineInfo is not expected to be placed into the eventPayload. Instead, the poLine entity is expected to be there under PO_LINE key, but there is no corresponding class in the module. Propose not to try to deserialize RelatedPoLineInfo from the PO_LINE value (despite deserialization doesn't fail) to avoid wrong interpretation about having RelatedPoLineInfo in the payload during the code reading/maintaining.

Reply:
thanks for suggestion, will re-open pr, after merging this one, because it contains code to fix kafka test issues on jenkins